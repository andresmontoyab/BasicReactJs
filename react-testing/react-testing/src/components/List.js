import React from 'react'

const List = ({title, list}) => {
    return (
        <div>
            <h1 className='big'>{title}</h1>
            <ul>
                {list.map(item => (
                    <li key={item.key}>
                        {item.name}
                    </li>
                ))}
            </ul>
        </div>
    )
}

export default List
