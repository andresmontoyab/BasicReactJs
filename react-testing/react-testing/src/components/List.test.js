import React from 'react'
import List from './List'
import {configure, shallow, mount, render} from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

configure({
    adapter: new Adapter
})


describe('Testing <List/> Component', () => {
    // given
    const fruits = [
        { name: 'Apple', id: 1},
        { name: 'Orange', id: 2},
        { name: 'Pineaple', id: 3},
        { name: 'Strawberry', id: 4},
    ]

    let wrapper

    beforeEach(() => {
        wrapper = shallow(<List title='Fruits' list={fruits}/>)

    })
    
    test('Checking nodes', () => {
        //then
        // Check if node exists
        expect(wrapper.find('h1').exists()).toBe(true)        

        // Check if it has css class
        expect(wrapper.find('h1').hasClass('big')).toBe(true)        

        // Check amount children
        expect(wrapper.find('ul').children().length).toBe(fruits.length)      
        
        // Validate content of node
        expect(wrapper.find('li').first().text()).toBe(fruits[0].name)      
        expect(wrapper.find('li').last().text()).toBe(fruits[fruits.length-1].name)      
        expect(wrapper.find('ul').childAt(2).text()).toBe(fruits[2].name)      
    })

    test('Validate update in props', () => {
        wrapper.setProps({
            list: [
                {
                    name: 'kiwi', id: 5
                }
            ]
        })

        //then
        expect(wrapper.find('li').length).toBe(1)

    })
    
    test('validate snapshot ', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
});

