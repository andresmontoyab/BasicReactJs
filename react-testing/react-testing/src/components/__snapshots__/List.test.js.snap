// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <List/> Component validate snapshot  1`] = `"<div><h1 class=\\"big\\">Fruits</h1><ul><li>Apple</li><li>Orange</li><li>Pineaple</li><li>Strawberry</li></ul></div>"`;
